File: backend/node/src/aws/rekognition.mjs
import { RekognitionClient, CompareFacesCommand, DetectTextCommand } from "@aws-sdk/client-rekognition";
import config from "../config.mjs";

const compareFaces = async (buff, path) => {
  const client = new RekognitionClient({
    region: config.rekognition.region,
    credentials: {
      accessKeyId: config.rekognition.accessKeyId,
      secretAccessKey: config.rekognition.secretAccessKey,
    },
  });

  const command = new CompareFacesCommand({
    SourceImage: {
      //imagen que el usuario sube para comparar
      Bytes: buff,
    },
    TargetImage: {
      // imagen que se encuentra en S3 del rostro como referencia
      S3Object: {
        Bucket: config.bucket,
        Name: path, //id de la imagen en S3
      },
    },
    SimilarityThreshold: 90, //umbral de similitud
  });

  try {
    const response = await client.send(command);
    return response;
  } catch (error) {
    console.error(error);
    return null;
  } finally {
    client.destroy();
  }
};

const extractText = async (buff) => {
    const client = new RekognitionClient({
      region: config.rekognition.region,
      credentials: {
        accessKeyId: config.rekognition.accessKeyId,
        secretAccessKey: config.rekognition.secretAccessKey,
      },
    });
  
    const comando = new DetectTextCommand({
      Image: {
        Bytes: buff,
      },
    });
  
    try {
      const response = await client.send(comando);
      console.log(response);
      return response;
    } catch (error) {
      console.error(error);
      return null;
    } finally {
      client.destroy();
    }
  };

export { compareFaces, extractText };

File: backend/python/src/aws/rekognition.py
from config import config
import boto3
from botocore.exceptions import NoCredentialsError, PartialCredentialsError

def compareFaces(buff, path):
    #Configuracion del cliente Rekognition
    rekognition_client = boto3.client(
        'rekognition',
        region_name=config['rekognition']['region'],
        aws_access_key_id=config['rekognition']['accessKeyId'],
        aws_secret_access_key=config['rekognition']['secretAccessKey']
    )

    try:
        response = rekognition_client.compare_faces(
            SourceImage={
                'Bytes': buff
            },
            TargetImage={
                'S3Object': {
                    'Bucket': config['bucket'],
                    'Name': path
                }
            },
            SimilarityThreshold=90 # Umbral de similitud
        )

        return response
    except (NoCredentialsError, PartialCredentialsError) as e:
        print("Error al comparar las caras" + str(e))
        return None
    except Exception as e:
        print("Error al comparar las caras" + str(e))
        return None